account_asset_management/models/account_asset.py

```diff
@@ -0,0 +1,1315 @@
+            if self.days_calc:
+                line_dates.append(stop_date)
+            else:
+                line_dates.append(line_date)
+
+        return line_dates
+
+    def _compute_depreciation_amount_per_fiscal_year(
+        self, table, line_dates, depreciation_start_date, depreciation_stop_date
+    ):
+        self.ensure_one()
+        currency = self.company_id.currency_id
+        fy_residual_amount = self.depreciation_base
+        i_max = len(table) - 1
+        asset_sign = self.depreciation_base >= 0 and 1 or -1
+        day_amount = 0.0
+        if self.days_calc:
+            days = (depreciation_stop_date - depreciation_start_date).days + 1
+            day_amount = self.depreciation_base / days
+
+        for i, entry in enumerate(table):
+            if self.method_time == "year":
+                year_amount = self._compute_year_amount(
+                    fy_residual_amount,
+                    depreciation_start_date,
+                    depreciation_stop_date,
+                    entry,
+                )
+                if self.method_period == "year":
+                    period_amount = year_amount
+                elif self.method_period == "quarter":
+                    period_amount = year_amount / 4
+                elif self.method_period == "month":
+                    period_amount = year_amount / 12
+                if i == i_max:
+                    if self.method in ["linear-limit", "degr-limit"]:
+                        fy_amount = fy_residual_amount - self.salvage_value
+                    else:
+                        fy_amount = fy_residual_amount
+                else:
+                    firstyear = i == 0 and True or False
+                    fy_factor = self._get_fy_duration_factor(entry, firstyear)
+                    fy_amount = year_amount * fy_factor
+                if (
+                    currency.compare_amounts(
+                        asset_sign * (fy_amount - fy_residual_amount), 0
+                    )
+                    > 0
+                ):
+                    fy_amount = fy_residual_amount
+                period_amount = currency.round(period_amount)
+                fy_amount = currency.round(fy_amount)
+            else:
+                fy_amount = False
+                if self.method_time == "number":
+                    number = self.method_number
+                else:
+                    number = len(line_dates)
+                period_amount = currency.round(self.depreciation_base / number)
+            entry.update(
+                {
+                    "period_amount": period_amount,
+                    "fy_amount": fy_amount,
+                    "day_amount": day_amount,
+                }
+            )
+            if self.method_time == "year":
+                fy_residual_amount -= fy_amount
+                if currency.is_zero(fy_residual_amount):
+                    break
+        i_max = i
```

Comment:
If the `table` variable is empty, it will not enter the loop; the variable `i` is uninitialized, and assigning it to `i_max` will trigger an error.